docs/src/integration_test_tool.md

```diff
@@ -1,3 +1,161 @@
+
+```mermaid
+graph TD
+    MyPkgMain -->|using| MyPkgA
+    MyPkgMain -->|using| MyPkgB
+    MyPkgMain -->|using| ExternDep1
+    MyPkgA -->|using| MyPkgC
+    MyPkgA -->|using| ExternDep2
+```
+
+The idea behind the package filter is that there are dependencies in the dependency graph that we cannot change directly. Let's assume we are the maintainer of all packages starting with `MyPkg` and the following dependency graph is given:
+
+```mermaid
+graph TD
+    MyPkgMain -->|using| MyPkgA
+    MyPkgMain -->|using| MyPkgB
+    MyPkgMain -->|using| ExternDep1 -->|using| MyPkgA
+    MyPkgA -->|using| MyPkgC
+    MyPkgA -->|using| ExternDep2
+```
+
+With the package filter `r"*"`, which means traverse all dependencies, the result would be `["MyPkgMain", "ExternDep1"]` if we are searching for `MyPkgA`. We are not the maintainer of `ExternDep1`, therefore we can only hardly introduce changes in `ExternDep1` and it make less sense to test `ExternDep1` with our code changes.
+
+With the package filter `r "*"`, which means that all dependencies are traversed, the result would be `["MyPkgMain", "ExternDep1"]` if we search for `MyPkgA`. We are not the maintainer of `ExternDep1`, so we can hardly change code in `ExternDep1` and it makes little sense to test `ExternDep1` with our code changes.
+
+## Package Filter: Special Cases
+
+### Dependencies of non matching packages
+
+In the following diagram, `MyPkgA` is not found with the package filter `r"^MyPkg*"`. The reason for this is that the traversal algorithm does not traverse `ExternDep1` and therefore does not check the dependencies of `ExternDep1`.
+
+```mermaid
+graph TD
+    MyPkgMain -->|using| ExternDep1 -->|using| MyPkgA
+```
+
+### The top graph package
+
+Sometimes the top graph package does not have the same naming scheme as all other packages. The section [Project Dependency Tree](#Project-Dependency-Tree) contains examples of this. If the top package is not included in the package filter, the entire graph is not traversed and the result is empty every time.
+
+```mermaid
+graph TD
+    MetaTestPkg -->|using| MyPkgMain -->|using| MyPkgA
+    MyPkgMain -->|using| MyPkgB
+```
+
+`depending_projects()` returns `[]` for the package filter `r"^MyPkg*"` and the package `MyPkgB`. The package filter `r"^MyPkg*|^MetaTestPkg$` returns `["MyPkgA"]`.
+
+# Project Dependency Tree
+
+We strongly recommend using a `project.toml` of a package from the package ecosystem as input for `depending_projects()` instead of defining the project dependency graph manually. The reason for this is that a manually defined dependency tree can easily become obsolete. In contrast, the `Project.toml` must be up to date, otherwise the Julia application will not work.
```

Comment:
```suggestion
We strongly recommend using a `Project.toml` of a package from the package ecosystem as input for `depending_projects()` instead of manually defining the project dependency graph. This is because a manually defined dependency tree can become obsolete. In contrast, the `Project.toml` must be up to date, otherwise, the Julia application will not work.
```